src/components/ButtonWithConfirmDialog/ButtonWithConfirmDialog.tsx

```diff
@@ -0,0 +1,74 @@
+import {useState, type ReactNode} from 'react';
+
+import {Button, type ButtonProps} from '@gravity-ui/uikit';
+
+import {CriticalActionDialog} from '../CriticalActionDialog';
+
+interface ButtonWithConfirmDialogProps<T, K> {
+    children: ReactNode;
+    onConfirmAction: () => Promise<T>;
+    onConfirmActionSuccess?: (() => Promise<K>) | VoidFunction;
+    dialogContent: string;
+    buttonDisabled?: ButtonProps['disabled'];
+    buttonView?: ButtonProps['view'];
+    buttonClassName?: ButtonProps['className'];
+}
+
+export function ButtonWithConfirmDialog<T, K>({
+    children,
+    onConfirmAction,
+    onConfirmActionSuccess,
+    dialogContent,
+    buttonDisabled = false,
+    buttonView = 'action',
+    buttonClassName,
+}: ButtonWithConfirmDialogProps<T, K>) {
+    const [isConfirmDialogVisible, setIsConfirmDialogVisible] = useState(false);
+    const [buttonLoading, setButtonLoading] = useState(false);
+
+    const handleConfirmAction = () => {
+        setButtonLoading(true);
+        return onConfirmAction();
+    };
+
+    const handleConfirmActionSuccess = async () => {
+        if (onConfirmActionSuccess) {
+            setButtonLoading(true);
+
+            try {
+                await onConfirmActionSuccess();
+            } catch {
+            } finally {
+                setButtonLoading(false);
+            }
+        }
+    };
+
+    const handleConfirmActionError = () => {
+        setButtonLoading(false);
+    };
+
+    return (
+        <>
```

Comment:
No need for wrapper

Reply:
It contains two components `CriticalActionDialog` and `Button`

Reply:
oops, overlooked